plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -121,18 +142,20 @@ func TestMetricsServerError(t *testing.T) {
 	defer mockPrometheus.Close()
 
 	logger := zap.NewNop()
+	tracer, exp, closer := tracerProvider(t)
+	defer closer()
 	address := mockPrometheus.Listener.Addr().String()
-	reader, err := NewMetricsReader(logger, config.Configuration{
+	reader, err := NewMetricsReader(config.Configuration{
 		ServerURL:      "http://" + address,
 		ConnectTimeout: defaultTimeout,
-	})
+	}, logger, tracer)
 	require.NoError(t, err)
-
 	m, err := reader.GetCallRates(context.Background(), &params)
 	assert.NotNil(t, m)
-
 	require.Error(t, err)
 	assert.Contains(t, err.Error(), "failed executing metrics query")
+	require.Len(t, exp.GetSpans(), 1, "HTTP request was traced and span reported")
+	assert.Equal(t, exp.GetSpans()[0].Name, "service_call_rate")
```

Comment:
`exp.GetSpans()[0].Status` is having null values in trace.Status fmt.

Reply:
the value comes up is `trace.Status(trace.Status{Code:0x0, Description:""})` which is null contained.

Reply:
that means you're not setting it correctly

Reply:
Again stuck. Not able to figure out what you said.

Reply:
when an error happens, your code needs to set span.Status. From what I see you're only doing span.RecordError, that does not set the status.